File: extensions/music/src/main/java/app/revanced/extension/music/patches/HideUpgradeButtonPatch.java
package app.revanced.extension.music.patches;

import app.revanced.extension.music.settings.Settings;

@SuppressWarnings("unused")
public class HideUpgradeButtonPatch {

    /**
     * Injection point
     */
    public static boolean hideUpgradeButton() {
        return Settings.HIDE_UPGRADE_BUTTON.get();
    }
}

File: patches/src/main/kotlin/app/revanced/patches/music/layout/upgradebutton/HideUpgradeButtonPatch.kt
package app.revanced.patches.music.layout.upgradebutton

import app.revanced.patcher.patch.bytecodePatch
import app.revanced.patches.music.layout.navigationbar.navigationBarPatch

@Deprecated("Patch is obsolete and was replaced by navigation bar patch", ReplaceWith("navigationBarPatch"))
@Suppress("unused")
val hideUpgradeButton = bytecodePatch{
    dependsOn(navigationBarPatch)
}

@Deprecated("Patch was renamed", ReplaceWith("hideUpgradeButton"))
@Suppress("unused")
val removeUpgradeButton = bytecodePatch{
    dependsOn(hideUpgradeButton)
}
